### PR TITLE
ci: let bors check GHA instead of Jenkins

### LINKED ISSUE #9136: Let bors check GHA instead of Jenkins
Once we are ready to make the switch from Jenkins to GHA, we need to configure bors to check GHA instead of Jenkins:
https://github.com/camunda/zeebe/blob/c39fd85b7d466a790d241b94623cf59fa38ee903/bors.toml#L1-L6

Our test workflow will be split over many jobs and we can't list all of them for bors to check. Instead, we can implement a workaround where we introduce a fake job that has a dependency on all other jobs and fails if any of them failed: 

https://github.com/rust-lang/crater/blob/9ab6f9697c901c4a44025cf0a39b73ad5b37d198/.github/workflows/bors.yml#L125-L149

### PR DESCRIPTION
**Pull request by lenaschoenburg**
closes #9136

### CODE REVIEW

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try

**Review by npepinpe — APPROVED** [review]
Looks good!

:wrench: As I mentioned, one other option is to have reusable workflows. The only downside is you can only have 1 layer, i.e. a workflow can call another, but that child workflow cannot call another workflow. So it may limit reusability.

Anyway, not a blocker, but it might be better in terms of mental model. Plus, you can execute the workflows individually if you need, or group them in different combinations. You can see an example here https://github.com/camunda-community-hub/zeebe-test-container/blob/main/.github/workflows/ci.yml

Plus, it's more maintainable I think since you don't need to keep track of all the jobs and make sure to keep them up to date.

Happy to pair on this if you want in a second iteration, or as part of this PR.

**Comment by npepinpe** [comment]
One small issue is still with the nodes not spawning, which seems worse this morning, and if not resolved will be ultra annoying, so I don't know if we want to postpone. I guess we can always change bors back to Jenkins temporarily.

**Comment by oleschoenburg** [comment]
bors try

**Comment by oleschoenburg** [comment]
bors try-

bors try

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    secrets: inherit
+  deploy:
+    name: Deploy
+    if: github.ref == 'refs/heads/main'
+    needs:
+      - code-quality
+      - test
+    uses: ./.github/workflows/deploy.yml
+    secrets: inherit
+  bors-check:
```
Comment (npepinpe) [inline .github/workflows/ci.yml:30]: :question: Couldn't bors just check for `Continuous Integration`? Or can it only check jobs and not workflows? I'm not super familiar what bors is looking at.

**Comment by npepinpe** [comment]
I'm wondering if we shouldn't hold off on this until after the retreat, when most of us are back. As we start relying on it, I expect we might have some effort to put to iron out some kinks. wdyt?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (oleschoenburg) [reply .github/workflows/ci.yml:30]: No, it can only check job, otherwise it'd be easy to just add a check for the test and code-quality workflows.

**Comment by oleschoenburg** [comment]
bors try

**Review by npepinpe — APPROVED** [review]
:+1:

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
I wonder if bors will use the new check here as well :thinking:

**Comment by oleschoenburg** [comment]
> I wonder if bors will use the new check here as well

The answer is yes. 13 minutes from `bors merge` to "Build succeeded" :sunglasses:

**Pull request merged**